tpke/src/ciphertext.rs

```diff
@@ -120,23 +120,27 @@ pub fn checked_decrypt_with_shared_secret<E: PairingEngine>(
     Some(decrypt_with_shared_secret(ciphertext, s))
 }
 
+fn blake2s_hash(input: &[u8]) -> Vec<u8> {
```

Comment:
We need to be careful with hash function selection, specially if there's stuff that has to be executed later on-chain. Although there's a precompile for the internal function F for Blake2 (https://eips.ethereum.org/EIPS/eip-152), we need to consider it. We don't want to end up with a hash function salad like it happened with our first Umbral implementation, where Blake2b, SHA256 and keccak256 had to coexist. In rust-umbral we switched to SHA256 for everything.

In any case, this is not a comment for this PR, but an issue to be opened.

Reply:
Tracking this issue here: https://github.com/nucypher/ferveo/issues/16